src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryHandles.java

```diff
@@ -178,7 +178,7 @@ public static VarHandle varHandle(ValueLayout layout) {
      * Java {@code int} to avoid dealing with negative values, which would be
      * the case if modeled as a Java {@code short}. This is illustrated in the following example:
      * <blockquote><pre>{@code
-    MemorySegment segment = MemorySegment.allocateNative(2, ResourceScope.newConfinedScope());
+    MemorySegment segment = MemorySegment.allocateNative(2, ResourceScope.newImplicitScope());
```

Comment:
Most changes in this patch are like this one: since `newConfinedScope()` is no longer automatically assigned a cleaner, we have to revert those changes back to what they were previously, by using an implicit scope instead. The difference compared to 17 is that the implicit scope is just a shared scope associated with a cleaner - it can be closed. There's no such concept of "special scope that can only be closed implicitly".